plugin-gradle/src/main/java/com/diffplug/gradle/spotless/JsonExtension.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016-2021 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.gradle.spotless;
+
+import javax.inject.Inject;
+
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.json.JsonFormatterStep;
+
+public class JsonExtension extends FormatExtension {
+	private static final Integer DEFAULT_INDENTATION = 4;
+	static final String NAME = "json";
+
+	@Inject
+	public JsonExtension(SpotlessExtension spotless) {
+		super(spotless);
+		addStep(createStep(DEFAULT_INDENTATION));
```

Comment:
None of the other `FormatExtension` add a step automatically, the user always picks one. That makes it easy for e.g. java users to swap `eclipseJdt` for `googleJavaFormat`, etc. In this case, users might want to swap for prettier or eclipseWtp in some cases.

Since there are so many ways to format json, we might want to rename your step to `JsonMinimalStep`, and add it to the built-in FormatExtension as `spotless { format 'json' { jsonMinimal().indentWithTabs()` or `jsonMinimal().indentWithSpaces(2)`.

The other option is to keep `JsonExtension`, and use it like `spotless { json { minimal().indentWithTabs()`. Your call whether to keep `JsonExtension` or to add your step into the parent `FormatExtension`.